### ISSUE TITLE
The connection in the TUN mode works only once

### ISSUE THREAD

**Issue by Bushroott** [InvisibleManVPN/InvisibleMan-XRayClient#126]
After launching the program, when you click the "start" button, administrator rights are requested, and a virtual network adapter is created in the system. After that, the connection is established, and everything works fine.
When you press the "stop" button, the virtual network adapter is removed from the system, and when you press the "start" button again, it no longer appears. At the same time, the program reports that the connection has been established, but in fact it is not there.
To reconnect, you need to exit the program and run it again.

OS Windows 7 x64
InvisibleMan-XRay v2.3.2 x32

**Comment by pps83** [comment]
@InvisibleManVPN seems like the last build is broken. I have the same issue where it's not working at all.
Do you have a vpn client without all that gui eye candy mess?

**Comment by pps83** [comment]
it's ridiculous that you can't even copy version info.
![image](https://github.com/user-attachments/assets/32598151-31d5-4128-a366-599ecfe60cf4)

**Comment by InvisibleManVPN** [comment]
Hey @Bushroott!
Did you check both x64 and x86 versions? In both versions, you have this issue?

**Comment by InvisibleManVPN** [comment]
@Bushroott If you can please download the previous version of [Invisible Man TUN v0.3.3](https://github.com/InvisibleManVPN/InvisibleMan-TUN/releases/tag/v0.3.3) and extract it in the `InvisibleManXRay-x86\TUN` path and check whether the program is working properly.

**Comment by Bushroott** [comment]
> @Bushroott If you can please download the previous version of [Invisible Man TUN v0.3.3](https://github.com/InvisibleManVPN/InvisibleMan-TUN/releases/tag/v0.3.3) and extract it in the `InvisibleManXRay-x86\TUN` path and check whether the program is working properly.

Unfortunately, the behavior of the program doesn`t change with the old driver either.

**Comment by Bushroott** [comment]
> Hey @Bushroott! Did you check both x64 and x86 versions? In both versions, you have this issue?

Both versions are affected.

**Comment by TemityStolyarov** [comment]
up
it's a little annoying to restart the app every time in order to reconnect to my VPN. is there a bugfix planned in the near future?

**Comment by InvisibleManVPN** [comment]
Hey @TemityStolyarov,
I'm working on it and for this reason, I postponed the new release to find a solution for that. Can I ask you to provide me a video about what you do in this process? Currently, this issue happens for some users and some others have no issue with that.

**Comment by i0FFLINE** [comment]
confirm issue

**Comment by InvisibleManVPN** [comment]
Hello @Bushroott @pps83 @TemityStolyarov @i0FFLINE 
I couldn't regenerate the problem. As I know, this bug happens to some users. Can you try regenerating the bug and sending me the result?
If you can, please download one of these zip files (based on the architecture of your OS):
- [InvisibleManTUN-x64.zip](https://github.com/user-attachments/files/17444881/InvisibleManTUN-x64.zip)
- [InvisibleManTUN-x86.zip](https://github.com/user-attachments/files/17444884/InvisibleManTUN-x86.zip)

In these files, I developed a logger system to log the process and record it into a `log.txt` file. So, we can track the things happens in this process.
After downloading, just go to the `{the path of the app}/TUN` and extract it there. Then, start the Invisible Man XRay and try to regenerate the problem. After that, just send me the `log.txt` file here.

**P.S:** It would be great if you could record a video of the process you do. It may help us to know more about the problem.

**Comment by TemityStolyarov** [comment]
this (InvisibleManTUN-x64.zip) file crashes after launching in a few seconds (runned as admin)
- 19-Oct-24 3:02:56 PM: Stop tunneling process.
![tunerr](https://github.com/user-attachments/assets/5ebb8196-78b1-4260-937d-6c2c8f1c8b55)

**Comment by InvisibleManVPN** [comment]
Hello @TemityStolyarov 
You should copy this file in the `{Invisible Man XRay path}/TUN` path and then open the **Invisible Man XRay.exe** app and try to click on the **RUN** button in **TUN** mode. You shouldn't open the **InvisibleMan-TUN.exe** directly.

**Comment by TemityStolyarov** [comment]
> You shouldn't open the **InvisibleMan-TUN.exe** directly

yep, my bad <details><summary>here is the contents</summary>

- 19-Oct-24 3:15:52 PM: Start tunneling process.
- 19-Oct-24 3:15:52 PM: Trying to cleanup profiles.
- 19-Oct-24 3:15:52 PM: Cleanup profiles successfully done.
- 19-Oct-24 3:15:52 PM: Trying to run tunnel.
- 19-Oct-24 3:15:52 PM: Wait until interface created.
- 19-Oct-24 3:15:52 PM: Set interface address.
- 19-Oct-24 3:15:52 PM: Wait until interface address set.
- 19-Oct-24 3:15:53 PM: Set interface DNS.
- 19-Oct-24 3:15:53 PM: Set routes.
- 19-Oct-24 3:16:02 PM: Stop tunneling process.
- 19-Oct-24 3:16:09 PM: Start tunneling process.
- 19-Oct-24 3:16:09 PM: Wait until interface created.
- 19-Oct-24 3:16:16 PM: Scheduler failed. Reason: Device with the name 'InvisibleMan-XRay' was not found.
- 19-Oct-24 3:16:16 PM: Start tunneling process failed. Reason: Device with the name 'InvisibleMan-XRay' was not found.
- 19-Oct-24 3:16:32 PM: Stop tunneling process.
- 19-Oct-24 3:16:34 PM: Stop tunneling process.
- 19-Oct-24 3:16:34 PM: Stop tunneling process.

</details>

[video link](https://www.youtube.com/watch?v=cSEcxGawbDA)

**Comment by InvisibleManVPN** [comment]
@TemityStolyarov Many thanks for your answer. Seems for some users, TUN can't detect that the tunneling process is finished (for the second time and so on). Can I ask you to open the **Command Prompt** and type this:

```
TASKLIST /FI "PID eq 12345"
```

I'm using the above command to ensure the tunneling process is finished and maybe the problem is here. Type it in your cmd and send me the result.

![image](https://github.com/user-attachments/assets/20d1b236-8fc1-44d1-aa09-378fbb57d575)

**Comment by i0FFLINE** [comment]
I suspect the cause is the Hyper-V installed on the machine. It has its own fancy ethernet interface(s).
If You have Windows 11 Pro - try to install and setup the second Virtual Switch with external connection type.
May be help.

**Comment by TemityStolyarov** [comment]
> ![image](https://private-user-images.githubusercontent.com/46280804/378163395-20d1b236-8fc1-44d1-aa09-378fbb57d575.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MTc1MDYsIm5iZiI6MTcyOTUxNzIwNiwicGF0aCI6Ii80NjI4MDgwNC8zNzgxNjMzOTUtMjBkMWIyMzYtOGZjMS00NGQxLWFhMDktMzc4ZmJiNTdkNTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDEzMjY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMGE1NjY3ZDk2Njc2MGI2ZTc1ODM2NWVmM2I2MjcyZjA4ZDljZGE1NjhiODVhODU5MGIzZDE1OWFiNjdiMjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dDxsTsjTUhuEgQhCY6Vqn_rwO0T4lGrrjN8XbTi_c0E)

same result for me: before the first connection, during use and after first disconnect

**Comment by pps83** [comment]
> I suspect the cause is the Hyper-V installed on the machine. It has its own fancy ethernet interface(s). If You have Windows 11 Pro - try to install and setup the second Virtual Switch with external connection type. May be help.

I had the issue with win10 and win11 and on both of them I have hyper-v enabled, and [softether vpn](https://www.softether.org/) installed as well.

**Comment by TemityStolyarov** [comment]
<details><summary>it worked</summary>

- 22-Oct-24 10:55:43 AM: Start tunneling process.
- 22-Oct-24 10:55:43 AM: Trying to cleanup profiles.
- 22-Oct-24 10:55:43 AM: Cleanup profiles successfully done.
- 22-Oct-24 10:55:43 AM: Trying to run tunnel.
- 22-Oct-24 10:55:43 AM: Wait until interface created.
- 22-Oct-24 10:55:44 AM: Set interface address.
- 22-Oct-24 10:55:44 AM: Wait until interface address set.
- 22-Oct-24 10:55:44 AM: Set interface DNS.
- 22-Oct-24 10:55:45 AM: Set routes.
- 22-Oct-24 10:55:55 AM: Stop tunneling process.
- 22-Oct-24 10:55:57 AM: Start tunneling process.
- 22-Oct-24 10:55:57 AM: Trying to cleanup profiles.
- 22-Oct-24 10:55:57 AM: Cleanup profiles successfully done.
- 22-Oct-24 10:55:57 AM: Trying to run tunnel.
- 22-Oct-24 10:55:57 AM: Wait until interface created.
- 22-Oct-24 10:55:57 AM: Set interface address.
- 22-Oct-24 10:55:57 AM: Wait until interface address set.
- 22-Oct-24 10:55:58 AM: Set interface DNS.
- 22-Oct-24 10:55:58 AM: Set routes.
- 22-Oct-24 10:56:09 AM: Stop tunneling process.

</details>

**Comment by InvisibleManVPN** [comment]
@TemityStolyarov Thank you too for testing and giving me feedback to find the problem.
@Bushroott @pps83 @i0FFLINE Guys please confirm that the issue is fixed for you as well. In the next release, I will push this fix.

**Comment by i0FFLINE** [comment]
works fine

**Comment by i0FFLINE** [comment]
@InvisibleManVPN 

but there's a nuance.
After connecting to the server, there is no traffic for a very long time. up to 1 minute no website responds or loads as at speeds of 10 kilobits/sec.

**Comment by InvisibleManVPN** [comment]
> but there's a nuance.
After connecting to the server, there is no traffic for a very long time. up to 1 minute no website responds or loads as at speeds of 10 kilobits/sec.

@i0FFLINE Sounds strange. Anyway, I don't think this problem is related to this issue. Didn't have this problem previously? Can you test more?

**Comment by i0FFLINE** [comment]
> > but there's a nuance.
> > After connecting to the server, there is no traffic for a very long time. up to 1 minute no website responds or loads as at speeds of 10 kilobits/sec.
> 
> @i0FFLINE Sounds strange. Anyway, I don't think this problem is related to this issue. Didn't have this problem previously? Can you test more?

my fault, its my provider laggy (i think router's uplink rebooted)
all works fine

**Comment by TemityStolyarov** [comment]
my friend reported a bug: he presses the 'Run' button and there is no connection. it just spins a little bit and immediately turns off. is this some other already known issue?

**Comment by InvisibleManVPN** [comment]
> my friend reported a bug: he presses the 'Run' button and there is no connection. it just spins a little bit and immediately turns off. is this some other already known issue?

@TemityStolyarov In the TUN mode or Proxy mode? To better track, please submit a new issue.

**Comment by InvisibleManVPN** [comment]
This issue resolved by did a fix on **Invisible Man TUN** service ([see this](https://github.com/InvisibleManVPN/InvisibleMan-TUN/pull/12))
In the next release of **Invisible Man XRay**, I will update the TUN service and this problem will be solved.
